### PR TITLE
[CDAP-11962][CDAP-11964] Undo & pipeline bugfixes

### PR DESCRIPTION
**Pull request by tonybach**
JIRAs:
- https://issues.cask.co/browse/CDAP-11962
- https://issues.cask.co/browse/CDAP-11964

### CODE REVIEW

File: cdap-ui/app/directives/complex-schema/complex-schema.js [hunk 1]
```diff
@@ -127,10 +127,13 @@ function ComplexSchemaController (avsc, SCHEMA_TYPES, $scope, uuid, $timeout, Sc
...
 
-    vm.formatOutput();
+    vm.formatOutput(true);
   }
 
-  vm.formatOutput = () => {
+  // In some cases, we edit the schema when the user opens a node, so the schema changes without the
+  // user doing anything. In those cases we should update the 'default' schema state to the state after
+  // we've done our initialzing i.e. updateDefault = true. Defaults to false.
+  vm.formatOutput = (updateDefault = false) => {
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/complex-schema/complex-schema.js:136]: Is `updateDefault` being used inside the function?

**Review by ajainarayanan — COMMENTED** [review]
Talked to @tonybach. Two things,

1. Rename converMacrosToRunTimeArguments. As this function acccepts macromap, user run time args and preferences and merges it into one.
2. Add a comment to `userRunTimeArgumentsForDisplay` in the store to distinguish it from `userRunTimeArguments`

File: cdap-ui/app/directives/complex-schema/complex-schema.js (hunk 1 above)
Comment (tonybach) [reply cdap-ui/app/directives/complex-schema/complex-schema.js:136]: Yes, it's used as a param of the `vm.parentFormatOutput` function that gets called inside this function.

**Review by ajainarayanan — COMMENTED** [review]
Code LGTM 👍

**Review by bdmogal — APPROVED** [review]

**Pull request merged**